app/models/scheme.rb

```diff
@@ -19,6 +19,8 @@ class Scheme < ApplicationRecord
 
   validate :validate_confirmed
 
+  before_validation :strip_whitespace
```

Comment:
Could just override `field=` methods... (has the advantage of setting it in memory the same as you'll save it already before actually doing the save, but might be unnecessary indirection...)

Reply:
How strongly do you feel about it?